engine/graphics/src/opengl/graphics_opengl.cpp

```diff
@@ -484,52 +484,51 @@ static void LogFrameBufferError(GLenum status)
 
     OpenGLContext* g_Context = 0x0;
 
-    static void UpdateGLHandle(HContext context, HOpenglID idx, GLuint handle)
+    static HOpenglID AddNewGLHandle(HContext _context, GLuint handle)
     {
-        OpenGLContext* gl_context = (OpenGLContext*)context;
-        gl_context->m_AllGLHandles[idx] = handle;
-    }
+        OpenGLContext* context = (OpenGLContext*) _context;
 
-    static HOpenglID AddNewGLHandle(HContext context, GLuint handle)
-    {
-        OpenGLContext* gl_context = (OpenGLContext*)context;
-        HOpenglID result_idx = gl_context->m_AllGLHandles.Size();
-        if (!gl_context->m_FreeIndexes.Empty())
+        ScopedLock lock(context->m_GLHandlesData.m_Mutex);
```

Comment:
I recommend using `DM_MUTEX_SCOPED_LOCK(lock)` so it's easier for us to do necessary changes if we need it. I consider the "ScopedLock" private.